ufs2arco/regrid/ufsregridder.py

```diff
@@ -0,0 +1,215 @@
+
+        >>> lats, lons = CICE6Regridder.compute_gaussian_grid(180, 360)
+
+        Open dataset using xarray or preferably :meth:`CICE6Dataset.open_dataset`
+
+        >>> ds = xr.open_mfdataset("./input/{ocn_,iceh_}*.nc")
+
+        Construct :class:`CICE6Regridder` object, specifying output grid lats & lons, dataset and config file
+
+        >>> rg = CICE6Regridder(lats, lons, ds, config_filename = "config-replay.yaml")
+
+        Regrid dataset
+
+        >>> ds = rg.regrid(ds)
+    """
+
+    def __init__(
+        self,
+        lats1d_out: np.array,
+        lons1d_out: np.array,
+        ds_in: xr.Dataset,
+        config_filename: str,
+        interp_method: str = "bilinear",
+    ) -> None:
+        """
+        Initialize the UFSRegridder object.
+        """
+
+        if not _has_xesmf:
+            raise ImportError(
+                f"Cannot import xesmf. Install with 'conda install -c conda-forge xesmf',"
+                f" and note the environment must be deactivated and reactivated"
+            )
+
+        super(UFSRegridder, self).__init__()
+        name = self.__class__.__name__
+
+        # Load configuration from YAML file
+        with open(config_filename, "r") as f:
+            self.config = yaml.safe_load(f).get(name, {})
+
+        # specify an output resolution
+        self.lats1d_out = lats1d_out
+        self.lons1d_out = lons1d_out
+        self.interp_method = interp_method
+
+        # create regridder
+        self._create_regridder(ds_in)
+
+    @abstractmethod
+    def _create_regridder(self, ds_in: xr.Dataset) -> None:
+        """
+        Create regridder: computes weights and creates three xesmf.Regridder instances.
+        Args:
+            ds_in (xarray.Dataset): xarray dataset needed for creating the regridder.
+        """
+
+    @staticmethod
+    def compute_gaussian_grid(nlat, nlon) -> (np.array, np.array):
+        """Compute gaussian grid latitudes and longitudes"""
+        latitudes, _ = gaussian_latitudes(nlat // 2)
+        longitudes = np.linspace(0, 360, nlon, endpoint=False)
+        return latitudes, longitudes
+
+    @staticmethod
+    def compute_latlon_grid(nlat, nlon) -> (np.array, np.array):
+        """Compute regular latlong grid coordinates"""
+        latitudes = np.linspace(-90, 90, nlat, endpoint=False)
+        longitudes = np.linspace(0, 360, nlon, endpoint=False)
+        return latitudes, longitudes
+
+    @staticmethod
+    def read_grid(
+        file, lats_name="grid_yt", lons_name="grid_xt"
+    ) -> (np.array, np.array):
+        """Read grid latitudes and longitudes from a netcdf file"""
+        ds = xr.open_dataset(file)
+        latitudes = ds[lats_name].values
+        longitudes = ds[lons_name].values
+        ds.close()
+        return latitudes, longitudes
+
+    def regrid_tripolar(
+        self,
+        ds_in: xr.Dataset,
+        ds_rot: xr.Dataset,
+        rg_tt: xe.Regridder,
+        rg_ut: xe.Regridder,
+        rg_vt: xe.Regridder,
```

Comment:
I think I know a way around the Windows build error. It is just having a problem with these lines, which are just typing. See my suggestions at the top of the module. If that gets rid of it, I say we roll with it and put off debugging Windows build on GHA for later...

Reply:
```suggestion
        rg_tt: Regridder,
        rg_ut: Regridder,
        rg_vt: Regridder,
```